pySDC/projects/DAE/sweepers/SemiImplicitDAE.py

```diff
@@ -0,0 +1,190 @@
+
+    .. math::
+        U = f(\vec{U}_0 + \Delta t (\mathbf{Q} \otimes \mathbf{I}_{n_d}) \vec{U}, \vec{z}, \tau),
+
+    .. math::
+        0 = g(\vec{U}_0 + \Delta t (\mathbf{Q} \otimes \mathbf{I}_{n_d}) \vec{U}, \vec{z}, \tau),
+
+    where
+    
+    - :math:`\tau=(\tau_1,..,\tau_M) in \mathbb{R}^M` the vector of collocation nodes,
+    - :math:`\vec{U}_0 = (u_0,..,u_0) \in \mathbb{R}^{MN_d}` the vector of initial condition spread to each node,
+    - spectral integration matrix :math:`\mathbf{Q} \in \mathbb{R}^{M \times M}`,
+    - :math:`\vec{U}=(U_1,..,U_M) \in \mathbb{R}^{MN_d}` the vector of unknown derivatives of differential variables
+      :math:`U_m \approx U(\tau_m) = u'(\tau_m) \in \mathbb{R}^{N_d}`,
+    - :math:`\vec{z}=(z_1,..,z_M) \in \mathbb{R}^{MN_a}` the vector of unknown algebraic variables
+      :math:`z_m \approx z(\tau_m) \in \mathbb{R}^{N_a}`,
+    - and identity matrix :math:`\mathbf{I}_{N_d} \in \mathbb{R}^{N_d \times N_d}`.
+
+    This sweeper treats the differential and the algebraic variables differently by only integrating the differential
+    components. Solving the nonlinear system, :math:`{U,z}` are the unknowns.
+
+    The sweeper implementation is based on the ideas mentioned in the KDC publication [1]_.
+
+    Parameters
+    ----------
+    params : dict
+        Parameters passed to the sweeper.
+
+    Attributes
+    ----------
+    QI : np.2darray
+        Implicit Euler integration matrix.
+
+    References
+    ----------
+    .. [1] J. Huang, J. Jun, M. L. Minion. Arbitrary order Krylov deferred correction methods for differential algebraic
+       equation. J. Comput. Phys. Vol. 221 No. 2 (2007).
+
+    Note
+    ----
+    The right-hand side of the problem DAE classes using this sweeper has to be exactly implemented in the way, the
+    semi-explicit DAE is defined. Define :math:`\vec{x}=(y, z)^T`, :math:`F(\vec{x})=(f(\vec{x}), g(\vec{x}))`, and the
+    matrix
+
+    .. math::
+        A = \begin{matrix}
+            I & 0 \\
+            0 & 0
+        \end{matrix}
+
+    then, the problem can be reformulated as
+
+    .. math::
+        A\vec{x}' = F(\vec{x}).
+
+    Then, setting :math:`F_{new}(\vec{x}, \vec{x}') = A\vec{x}' - F(\vec{x})` defines a DAE of fully-implicit form
+
+    .. math::
+        0 = F_{new}(\vec{x}, \vec{x}').
+
+    Hence, the method ``eval_f`` of problem DAE classes of semi-explicit form implements the right-hand side in the way of
+    returning :math:`F(\vec{x})`, whereas ``eval_f`` of problem classes of fully-implicit form return the right-hand side
+    :math:`F_{new}(\vec{x}, \vec{x}')`.
+    """
+
+    def __init__(self, params):
+        """Initialization routine"""
+
+        if 'QI' not in params:
+            params['QI'] = 'IE'
+
+        # call parent's initialization routine
+        super().__init__(params)
+
+        msg = f"Quadrature type {self.params.quad_type} is not implemented yet. Use 'RADAU-RIGHT' instead!"
+        if self.coll.left_is_node:
```

Comment:
Shouldn't this be `if not self.coll.right_is_node:`? I assume you don't want the collocation update..?

Reply:
We also wouldn't like to use ``'LOBATTO'`` since it is not tested yet and for ``'LOBATTO'`` an initial condition for ``du`` is needed.